dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1016,26 +1019,57 @@ private String getEmail(GHMyself myself) throws IOException {
      * Updates a user object with metadata from GitHub
      * @param user the user to be updated
      */
-    public void syncUserMetadataFromGitHub(User user) {
+    public void syncUserMetadataFromGitHub(User user, TokenDAO tokenDAO, boolean updateToken) {
         // eGit user object
         try {
             GHMyself myself = github.getMyself();
-            User.Profile profile = new User.Profile();
-            profile.name = myself.getName();
+            User.Profile profile = getProfile(user, myself);
             profile.email = getEmail(myself);
-            profile.avatarURL = myself.getAvatarUrl();
-            profile.bio = myself.getBlog();  // ? not sure about this mapping in the new api
-            profile.location = myself.getLocation();
-            profile.company = myself.getCompany();
-            profile.username = myself.getLogin();
-            Map<String, User.Profile> userProfile = user.getUserProfiles();
-            userProfile.put(TokenType.GITHUB_COM.toString(), profile);
-            user.setAvatarUrl(myself.getAvatarUrl());
+
+            // Update token. Username on GitHub could have changed and need to collect the GitHub user id as well
+            if (updateToken) {
+                Token usersGitHubToken = tokenDAO.findGithubByUserId(user.getId()).get(0);
+                usersGitHubToken.setOnlineProfileId(profile.onlineProfileId);
+                usersGitHubToken.setUsername(profile.username);
+            }
         } catch (IOException ex) {
             LOG.info("Could not find user information for user " + user.getUsername(), ex);
         }
     }
 
+    // This function has no use outside of gathering user's GitHub IDs the first time. This uses the GitHub token of the admin user calling the new, one-time-use endpoint.
+    // This will attempt to get the GitHub profile info (including id) of users we were unable to get by calling the github.getMyself() function above.
+    public void syncUserMetadataFromGitHubByUsername(User user, TokenDAO tokenDAO) {
+        // eGit user object
+        try {
+            GHUser ghUser = github.getUser(user.getUserProfiles().get(TokenType.GITHUB_COM.toString()).username);
```

Comment:
Just to confirm, if this fails, e.g., the profile doesn't exist, it throws an IOException?

Reply:
No, if the user profile doesn't exist it throws the NPE which will get caught here https://github.com/dockstore/dockstore/pull/4204/files#diff-2c8f98ad7fbc26776c9437aeb9dfe12d07fe60de2ada259bf78564e92198295aR765

Reply:
I shouldn't have said `profile`. I meant `gitHub.getUser`, which I just looked at, and does throw a `GHFileNotFoundException`, which is an `IOException`, if the user doesn't exist on GitHub.

But now your response get me looking at the `user.getUserProfiles().get` which does potentially return the null. I feel it's better for code readability to explicitly test for it rather than rely on a catch in the calling function. What if somebody else calls this and isn't aware that it throws an Exception?

Reply:
Okay, I will explicitly check that `user.getUserProfiles().get` isn't null, I think that makes sense. But, I'd also say this function should be deleted in a hotfix. I don't think anyone should be getting a github user this way and this is just something we're forced to do this one time.